File: 2022/04/04.py
import sys

lines = sys.stdin.read().strip().split("\n")

score1 = 0
score2 = 0
for line in lines:
    l1, r1, l2, r2 = line.replace(",", "-").split("-")

    if (l1 <= l2 and r1 >= r2) or (l2 <= l1 and r2 >= r1):
        score1 += 1

    if (r2 >= l1 >= l2) or (r2 >= r1 >= l2) or (r1 >= l2 >= l1) or (r1 >= r2 >= l1):
        score2 += 1

print(score1)
print(score2)

File: 2022/04/04.kt
fun main() {
    val list = generateSequence(::readlnOrNull)
        .map { it.replace(",", "-").split("-") }
        .map { nums -> nums.map { it.toInt() } }
        .toList()

    var score1 = 0
    var score2 = 0
    for ((l, r, l2, r2) in list) {
        if ((l <= l2 && r >= r2) || (l >= l2 && r <= r2)) score1 += 1
        if ((l..r intersect l2..r2).isNotEmpty()) score2 += 1
    }

    println(score1)
    println(score2)

    assert(score1 == 490) { "First answer is wrong." }
    assert(score2 == 921) { "Second answer is wrong." }
}